velox/dwio/common/MetadataFilter.cpp

```diff
@@ -163,19 +164,19 @@ std::unique_ptr<MetadataFilter::Node> MetadataFilter::Node::fromExpression(
   if (!call) {
     return nullptr;
   }
-  if (call->name() == "and") {
+  if (call->name() == expression::kAnd) {
     auto lhs = fromExpression(*call->inputs()[0], evaluator, negated);
     auto rhs = fromExpression(*call->inputs()[1], evaluator, negated);
     return negated ? OrNode::create(std::move(lhs), std::move(rhs))
                    : AndNode::create(std::move(lhs), std::move(rhs));
   }
-  if (call->name() == "or") {
+  if (call->name() == expression::kOr) {
     auto lhs = fromExpression(*call->inputs()[0], evaluator, negated);
     auto rhs = fromExpression(*call->inputs()[1], evaluator, negated);
     return negated ? AndNode::create(std::move(lhs), std::move(rhs))
                    : OrNode::create(std::move(lhs), std::move(rhs));
   }
-  if (call->name() == "not") {
+  if (call->name() == expression::bug::kNot) {
```

Comment:
This one we maybe need a singleton to retrieve the prefix.  Or we resolve the name (we have evaluator here) to make sure it is indeed NOT function

Reply:
https://github.com/facebookincubator/velox/blob/09d2f0c724b2e7c3325dc5c6bad34a16297cd374/velox/connectors/hive/HiveConnectorUtil.cpp#L818